misc-scripts/xref_mapping/XrefParser/ZFINParser.pm

```diff
@@ -45,68 +46,78 @@ sub run {
   my $file = @{$files}[0];
   my $dir = dirname($file);
 
-  my (%swiss) = %{$self->get_valid_codes("uniprot/",$species_id, $dbi)};
+  my (%swiss) = %{$self->get_valid_codes("uniprot/swissprot",$species_id, $dbi)};
   my (%refseq) = %{$self->get_valid_codes("refseq",$species_id, $dbi)};
 
   my $swissprot_io =
     $self->get_filehandle( catfile( $dir, 'uniprot.txt' ) );
 
   if ( !defined $swissprot_io ) {
-    print STDERR "ERROR: Could not open " . catfile( $dir, 'uniprot.txt' ). "\n" ;
+    croak "ERROR: Could not open " . catfile( $dir, 'uniprot.txt' ). "\n" ;
     return 1;    # 1 error
   }
 
-#e.g.
-#ZDB-GENE-000112-30      couptf2 O42532
-#ZDB-GENE-000112-32      couptf3 O42533
-#ZDB-GENE-000112-34      couptf4 O42534
 
+  my $swissprot_csv = Text::CSV->new({
+      sep_char       => "\t",
+      empty_is_undef => 1,
+      strict         => 1,
+  }) or croak "Could not use swissprot file $file: " . Text::CSV->error_diag();
+
+  $swissprot_csv->column_names([ 'zfin',
+                                 'so',
+                                 'label',
+                                 'acc',
+  ]);
 
   my %description;
 
   my $sql = "insert ignore into synonym (xref_id, synonym) values (?, ?)";
```

Comment:
the custom sql should be dropped and methods from the (here) base parser used.
In this case BaseParser::add_to_syn() provides this functionality.

Reply:
This statement handle never gets executed anywhere, as it turns out. I suspect it never got cleaned up when the add_to_syn_for_mult_sources was added further down the file()